src/petals/client/routing/sequence_manager.py

```diff
@@ -0,0 +1,256 @@
+                    self._rpc_info = MSGPackSerializer.loads(outputs.serialized_info)
+                    break
+                except Exception as e:
+                    delay = self.get_retry_delay(attempt_no)
+                    logger.warning(
+                        f"Caught exception when gathering information from peer {peer_id} "
+                        f"(retry in {delay:.0f} sec): {repr(e)}"
+                    )
+                    traceback_level = logging.DEBUG if str(e) else logging.WARNING
+                    logger.log(traceback_level, "See detailed traceback below:", exc_info=True)
+                    time.sleep(delay)
+
+        return self._rpc_info
+
+    def get_retry_delay(self, attempt_no: int) -> float:
+        if attempt_no == 0:
+            return 0
+        return self.min_backoff * 2 ** (attempt_no - 1)
+
+    def get_request_metadata(self, protocol: str, *args, **kwargs) -> Optional[bytes]:
+        """
+        :param protocol: one of "rpc_forward", "rpc_backward" or "rpc_inference"
+        :param args: request-specific inputs, typicall block uids and input tensors
+        :param kwargs: additional request context, such as remote peer ID
+        :returns: msgpack-serialized metadata dict that will be passed alongside a given request
+        """
+        return MSGPackSerializer.dumps(dict(points=self.policy.get_points(protocol, *args, **kwargs)))
+
+    def shutdown(self):
+        self._thread.shutdown()
+
+
+class _SequenceManagerUpdateThread(threading.Thread):
+    def __init__(self, update_period: float, ref_update_manager: WeakMethod):
+        super().__init__(daemon=True)
+        self.ref_update_manager = ref_update_manager
+        self.ready = threading.Event()
+        self.trigger = threading.Event()
+        self.last_update_time = -float("inf")
+        self.update_period = update_period
+        self.should_shutdown = False
+
+    def run(self) -> None:
+        while not self.should_shutdown:
+
+            self.trigger.wait(max(0.0, min(self.update_period, time.perf_counter() - self.last_update_time)))
+
+            if self.should_shutdown:
+                logger.debug(f"{self.__class__.__name__} is shutting down")
+                break
+
+            if not self.trigger.is_set() and time.perf_counter() - self.last_update_time >= self.update_period:
+                continue  # waited for update_period, but found that our info was already updated in the meantime
+
+            update_manager = self.ref_update_manager()
+            if update_manager is None:
+                logger.debug(f"{self.__class__.__name__} exited because the sequence manager no longer exists")
+                break
+
+            try:
+                update_manager()
+                self.trigger.clear()
+            except Exception as e:
+                logger.exception(e)
+            finally:
+                del update_manager
+
+        logger.info(f"{self.__class__.__name__} thread exited")
```

Comment:
```suggestion
        logger.debug(f"{self.__class__.__name__} thread exited")
```

(not useful to the end user)